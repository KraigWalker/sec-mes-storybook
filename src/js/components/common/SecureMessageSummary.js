import React from 'react';
import GetIcon from './GetIcon';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import cx from 'classnames';
import MessageEntity from '../../entities/MessageEntity';
import ModalComponent from '../common/ModalComponent';
import { updateMessageData} from '../../actions/AppActions';
import SendMessageRequestEntity from '../../entities/SendMessageRequestEntity.js'
import { connect } from 'react-redux';
let messageEntity = new SendMessageRequestEntity();
class SecureMessageSummary extends React.Component {
    constructor(props) {
        super(props);
        this.handleDelete = this.handleDelete.bind(this);
        this.state = {
            showDeleteConfirmModal: false,
            showDeleteSuccessModal: false,
        };
        this.returnModalComponent = this.returnModalComponent.bind(this);
        this.closeModal = this.closeModal.bind(this);
        this.deleteClick = this.deleteClick.bind(this);
        this.returnDeleteSuccessModalComponent = this.returnDeleteSuccessModalComponent.bind(this);
        this.closeSuccessModal = this.closeSuccessModal.bind(this);
    }
    
    getSummaryIcon = () => {
        const { message } = this.props;
        let iconId = message.status === 'READ' ? 'icon-message-open' : 'icon-envelope';
        return (<span className="c-message__icon"><GetIcon id={iconId} width="24px" height="24px" /></span>);
    }
    hasOnClick = () => {
        const { message } = this.props;
        let path = message.status === 'DRAFT' ? `${window.baseURl}/draftsecuremessage` : `${window.baseURl}/viewmessage`;
        let messageTitle =''    ;
        if (message.status === 'NEW') {
            messageTitle = `Unread ` + message.getSubject();
        } else {            
            messageTitle = message.getSubject();
        }
        if (!this.props.threadFlag) {
            if (message.status === 'READ') {
                return (<Link to={{ pathname: path, messageDetail: message }} className="c-message__summary__head__title__subject__link">{message.getSubject()}</Link>);
            } else {
                return (<Link aria-label={`${messageTitle}`} to={{ pathname: path, messageDetail: message }} className="c-message__summary__head__title__subject__link">{message.getSubject()}</Link>);        
            }
        } else {
            return message.getSubject();
        }
    }
    getReplyButton = () => {
        const { message } = this.props;
        let replymessage = ``;
        if (message.status === 'READ') {
            replymessage = `Reply ` + message.getSubject();
        } else {
            replymessage = `Reply Unread ` + message.getSubject();        
        }
        return (!this.props.threadFlag) && 
            (<Link to={{ pathname: `${window.baseURl}/replysecuremessage`, backPath: this.props.viewMessageFlag ? `${window.baseURl}/viewmessage` : `${window.baseURl}/securemessages`, messageDetail: message }} className="c-btn c-btn--link c-message__summary__head__actions__reply u-no-padding">
                <span id="replyMsg" className="c-message__summary__head__actions__reply__txt" aria-label={`${replymessage}`}>Reply</span>
                <span className="c-message__summary__head__actions__reply__icon">
                    <GetIcon id="icon-reply" width="24px" height="24px" />
                </span>
            </Link>)
    }

    getDeleteButton = () => {
        const { message } = this.props;
        let deletemessage = ``;
       if (!message.status !== 'NEW' ) {
            deletemessage = `Delete ` + message.getSubject();
        } else {
            deletemessage = `Delete Unread ` + message.getSubject();   
        }
        return (<button className="c-btn c-btn--link c-message__summary__head__actions__delete u-no-padding" onClick={this.handleDelete}>
            <span id="deleteMsg" className="c-message__summary__head__actions__delete__txt" aria-label={`${deletemessage}`}>Delete</span>
            <span className="c-message__summary__head__actions__delete__icon">
                <GetIcon id="icon-delete" width="24px" height="24px" />
            </span>
        </button>);
    }
    handleDelete(data) {
        this.setState({ showDeleteConfirmModal: true });
    }
    closeModal() {
        setTimeout(() => {
            document.getElementById('headingTag').focus();
        }, 20);
        this.setState({ showDeleteConfirmModal: false });
    }
    deleteClick() {
        this.setState({ showDeleteSuccessModal: true, showDeleteConfirmModal: false });
        this.props.dispatch(updateMessageData(this.props.message, this.props.message.id, "DELETED"));
    }
    closeSuccessModal(){
       // document.getElementById('headingTag').focus();
        this.setState({ showDeleteSuccessModal: false });
    }
    returnDeleteSuccessModalComponent() {
        let bodyContent = <div><div><GetIcon id="icon-success" width="68px" height="68px" /></div>Message Deleted</div>;
        let footerButtons = <div><button type="button" onClick={this.closeSuccessModal} className="c-btn c-btn--primary c-btn--sm c-modal__button">OK</button></div>;
        return (<ModalComponent show
            onHide={this.closeSuccessModal}
            customClass={"c-modal c-modal--center"}
            bsSize={'small'}
            modalheading={''}
            modalbody={bodyContent}
            modalfooter={footerButtons}
            modalInContainer={false}
            closeButton />);
    }
    returnModalComponent() {
        let bodyContent = <div className="callout callout__error">You wonâ€™t be able to recover this message if you delete it.</div>;
        let footerButtons = <div><button type="button" onClick={this.closeModal} className="c-btn c-btn--secondary c-modal__button">Close</button>
            <button type="button" onClick={this.deleteClick} className="c-btn c-btn--default c-modal__button">Delete message</button></div>;
        return (<ModalComponent show
            onHide={this.closeModal}
            customClass={"c-modal"}
            modalheading={'Delete this message?'}
            modalbody={bodyContent}
            modalfooter={footerButtons}
            modalInContainer={false}
            closeButton />);
    }
    render() {
        const { message } = this.props;
        let messageClass = cx({
            'c-message': true,
            'c-message--stacked': this.props.listFlag,
            'c-message--read': message.status === 'READ',
            'u-position-relative': !this.props.threadFlag,
            'c-message--noborder': this.props.threadFlag,
        });
        let summaryClass = cx({
            'c-message__summary': true,
            'c-message__summary--no-icon': message.status === 'DRAFT' || message.status === 'SENT' || message.status === 'PENDING',
        });
        let titleClass = cx({
            'c-message__summary__head__title': true,
            'c-message__summary__head__title--draft': message.status === 'DRAFT',
        });
        let subjectClass = cx({
            'c-message__summary__head__title__subject': true,
            'c-message__summary__head__title__subject--read': message.status !== 'NEW',
        });
        let actionsClass = cx({
            'c-message__summary__head__actions': true,
            'u-position-relative': !this.props.threadFlag,
        });


        return (
            <div className={messageClass}>
                {(message.status === 'READ' || message.status === 'NEW') && this.getSummaryIcon()}
                <div className={summaryClass} >
                    <div className="c-message__summary__head">
                        <div className={titleClass}>
                            <h2 id="subjectMsg" className={subjectClass}>
                                {this.hasOnClick()}
                            </h2>
                            <p className="c-message__summary__head__title__ref">{message.getReference()}</p>
                        </div>
                        <div className={actionsClass}>
                            {(message.status === 'NEW' || message.status === 'READ') && this.getReplyButton(message)}
                            {!this.props.threadFlag && this.getDeleteButton()}
                        </div>
                    </div>
                    {!this.props.viewMessageFlag && <p className="c-message__summary__account">{message.getMessageBody()}</p>}
                    {this.props.viewMessageFlag && <p className="c-message__summary__account">Account : {this.props.message.account.accountNumber}</p>}
                    <p className="c-message__summary__date">{message.getDateCreated()}</p>
                </div>
                {this.state.showDeleteConfirmModal && this.returnModalComponent()}
                {this.state.showDeleteSuccessModal && this.returnDeleteSuccessModalComponent()}
            </div>
        );
    }
}

SecureMessageSummary.propTypes = {
    listFlag: PropTypes.bool,
    draftFlag: PropTypes.bool,
    readFlag: PropTypes.bool,
    hasOnClick: PropTypes.bool,
    threadFlag: PropTypes.bool,
    sentFlag: PropTypes.bool,
    message: PropTypes.instanceOf(MessageEntity),
};
SecureMessageSummary.defaultProps = {
    listFlag: false,
    draftFlag: false,
    readFlag: false,
    hasOnClick: false,
    threadFlag: false,
    sentFlag: false,
};
/**
 * Maps the state of the component to the state of the redux store
 * @param {object} state. State of the application
 */
const mapState = (state) => {
    return {
        messagesubjects: state.subjects,
        messageaccounts: state.accounts,
    }
};
export default connect(mapState)(SecureMessageSummary);
