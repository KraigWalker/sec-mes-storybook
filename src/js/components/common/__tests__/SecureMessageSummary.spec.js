
import React from 'react';
import { shallow, mount } from 'enzyme';
import { SecureMessageSummary } from '../SecureMessageSummary';

const setup = (setupProps) => {
  const props = {
    content: {
      back: 'Back',
      noSpecificAccount: 'noSpecificAccount'
    },
    messageDetail: {
      newMessageError: true,
      delSuccessModal: true,
    },
    message: {
      status: 'READ',
      getSubject: jest.fn(),
      getReference: jest.fn(),
      getMessageBody: jest.fn(),
      getDateCreated: jest.fn(),
      account: {
        number: '32236',
      }
    },
    messages: {
      draftError: true,
    },
    dispatch: jest.fn(),
    ...setupProps
  }
  const wrapper = shallow(<SecureMessageSummary {...props} />);
  return {
    wrapper,
    props
  }
}


describe("SecureMessageSummary snapshot", () => {
  const dispatch = jest.fn();
  const getSubject = jest.fn();
  const getReference = jest.fn();
  const getMessageBody = jest.fn();
  const getDateCreated = jest.fn();
  let props = {
    content: {
      back: 'Back',
      noSpecificAccount: 'noSpecificAccount'
    },
    viewMessageFlag: true,
    messageDetail: {
      newMessageError: true,
      delSuccessModal: true,
    },
    message: {
      status: 'NEW',
      getSubject: getSubject,
      getReference: getReference,
      getMessageBody: getMessageBody,
      getDateCreated: getDateCreated,
      account: {
        number: '32236',
      }
    },
    messages: {
      draftError: true,
    },
    dispatch: dispatch
  };
  let component = shallow(<SecureMessageSummary {...props} />);
  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  describe("SecureMessageSummary snapshot", () => {
    const dispatch = jest.fn();
    const getSubject = jest.fn();
    const getReference = jest.fn();
    const getMessageBody = jest.fn();
    const getDateCreated = jest.fn();
    let props = {
      content: {
        back: 'Back',
        noSpecificAccount: 'noSpecificAccount'
      },
      messageDetail: {
        newMessageError: true,
        delSuccessModal: true,
      },
      message: {
        status: 'READ',
        getSubject: getSubject,
        getReference: getReference,
        getMessageBody: getMessageBody,
        getDateCreated: getDateCreated,
        account: {
          number: '32236',
        }
      },
      messages: {
        draftError: true,
      },
      dispatch: dispatch
    };
    let component = shallow(<SecureMessageSummary {...props} />);
    it('should match to snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe("SecureMessageSummary snapshot", () => {
    const dispatch = jest.fn();
    const getSubject = jest.fn();
    const getReference = jest.fn();
    const getMessageBody = jest.fn();
    const getDateCreated = jest.fn();
    let props = {
      content: {
        back: 'Back',
        noSpecificAccount: 'noSpecificAccount'
      },
      messageDetail: {
        newMessageError: true,
        delSuccessModal: true,
      },
      message: {
        status: 'DRAFT',
        getSubject: getSubject,
        getReference: getReference,
        getMessageBody: getMessageBody,
        getDateCreated: getDateCreated,
        account: {
          number: undefined,
        }
      },
      messages: {
        draftError: true,
      },
      dispatch: dispatch
    };
    let component = shallow(<SecureMessageSummary {...props} />);
    it('should match to snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});