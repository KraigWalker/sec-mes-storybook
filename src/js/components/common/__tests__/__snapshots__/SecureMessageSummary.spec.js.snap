// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecureMessageSummary snapshot SecureMessageSummary snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SecureMessageSummary
    content={
      Object {
        "back": "Back",
        "noSpecificAccount": "noSpecificAccount",
      }
    }
    dispatch={[MockFunction]}
    message={
      Object {
        "account": Object {
          "number": "32236",
        },
        "getDateCreated": [MockFunction],
        "getMessageBody": [MockFunction],
        "getReference": [MockFunction],
        "getSubject": [MockFunction],
        "message": "Hello there",
        "status": "READ",
      }
    }
    messageDetail={
      Object {
        "delSuccessModal": true,
        "newMessageError": true,
      }
    }
    messages={
      Object {
        "draftError": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithTheme(MailItem)
        archiveOnClick={null}
        content={
          Object {
            "back": "Back",
            "noSpecificAccount": "noSpecificAccount",
          }
        }
        deleteOnClick={null}
        dispatch={[MockFunction]}
        id={undefined}
        isArchived={false}
        isRead={true}
        mailDate={undefined}
        mailOnClick={[Function]}
        mailSubject="undefined "
        mailSummary="Hello there"
        message={
          Object {
            "account": Object {
              "number": "32236",
            },
            "getDateCreated": [MockFunction],
            "getMessageBody": [MockFunction],
            "getReference": [MockFunction],
            "getSubject": [MockFunction],
            "message": "Hello there",
            "status": "READ",
          }
        }
        messageDetail={
          Object {
            "delSuccessModal": true,
            "newMessageError": true,
          }
        }
        messages={
          Object {
            "draftError": true,
          }
        }
        moveToInboxOnClick={null}
        replyOnClick={null}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "archiveOnClick": null,
        "content": Object {
          "back": "Back",
          "noSpecificAccount": "noSpecificAccount",
        },
        "deleteOnClick": null,
        "dispatch": [MockFunction],
        "id": undefined,
        "isArchived": false,
        "isRead": true,
        "mailDate": undefined,
        "mailOnClick": [Function],
        "mailSubject": "undefined ",
        "mailSummary": "Hello there",
        "message": Object {
          "account": Object {
            "number": "32236",
          },
          "getDateCreated": [MockFunction],
          "getMessageBody": [MockFunction],
          "getReference": [MockFunction],
          "getSubject": [MockFunction],
          "message": "Hello there",
          "status": "READ",
        },
        "messageDetail": Object {
          "delSuccessModal": true,
          "newMessageError": true,
        },
        "messages": Object {
          "draftError": true,
        },
        "moveToInboxOnClick": null,
        "replyOnClick": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithTheme(MailItem)
          archiveOnClick={null}
          content={
            Object {
              "back": "Back",
              "noSpecificAccount": "noSpecificAccount",
            }
          }
          deleteOnClick={null}
          dispatch={[MockFunction]}
          id={undefined}
          isArchived={false}
          isRead={true}
          mailDate={undefined}
          mailOnClick={[Function]}
          mailSubject="undefined "
          mailSummary="Hello there"
          message={
            Object {
              "account": Object {
                "number": "32236",
              },
              "getDateCreated": [MockFunction],
              "getMessageBody": [MockFunction],
              "getReference": [MockFunction],
              "getSubject": [MockFunction],
              "message": "Hello there",
              "status": "READ",
            }
          }
          messageDetail={
            Object {
              "delSuccessModal": true,
              "newMessageError": true,
            }
          }
          messages={
            Object {
              "draftError": true,
            }
          }
          moveToInboxOnClick={null}
          replyOnClick={null}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "archiveOnClick": null,
          "content": Object {
            "back": "Back",
            "noSpecificAccount": "noSpecificAccount",
          },
          "deleteOnClick": null,
          "dispatch": [MockFunction],
          "id": undefined,
          "isArchived": false,
          "isRead": true,
          "mailDate": undefined,
          "mailOnClick": [Function],
          "mailSubject": "undefined ",
          "mailSummary": "Hello there",
          "message": Object {
            "account": Object {
              "number": "32236",
            },
            "getDateCreated": [MockFunction],
            "getMessageBody": [MockFunction],
            "getReference": [MockFunction],
            "getSubject": [MockFunction],
            "message": "Hello there",
            "status": "READ",
          },
          "messageDetail": Object {
            "delSuccessModal": true,
            "newMessageError": true,
          },
          "messages": Object {
            "draftError": true,
          },
          "moveToInboxOnClick": null,
          "replyOnClick": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`SecureMessageSummary snapshot SecureMessageSummary snapshot should match to snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SecureMessageSummary
    content={
      Object {
        "back": "Back",
        "noSpecificAccount": "noSpecificAccount",
      }
    }
    dispatch={[MockFunction]}
    message={
      Object {
        "account": Object {
          "number": undefined,
        },
        "getDateCreated": [MockFunction],
        "getMessageBody": [MockFunction],
        "getReference": [MockFunction],
        "getSubject": [MockFunction],
        "message": "Hello there",
        "status": "DRAFT",
      }
    }
    messageDetail={
      Object {
        "delSuccessModal": true,
        "newMessageError": true,
      }
    }
    messages={
      Object {
        "draftError": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithTheme(MailItem)
        archiveOnClick={null}
        content={
          Object {
            "back": "Back",
            "noSpecificAccount": "noSpecificAccount",
          }
        }
        deleteOnClick={null}
        dispatch={[MockFunction]}
        id={undefined}
        isArchived={false}
        isRead={true}
        mailDate={undefined}
        mailOnClick={[Function]}
        mailSubject="undefined "
        mailSummary="Hello there"
        message={
          Object {
            "account": Object {
              "number": undefined,
            },
            "getDateCreated": [MockFunction],
            "getMessageBody": [MockFunction],
            "getReference": [MockFunction],
            "getSubject": [MockFunction],
            "message": "Hello there",
            "status": "DRAFT",
          }
        }
        messageDetail={
          Object {
            "delSuccessModal": true,
            "newMessageError": true,
          }
        }
        messages={
          Object {
            "draftError": true,
          }
        }
        moveToInboxOnClick={null}
        replyOnClick={null}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "archiveOnClick": null,
        "content": Object {
          "back": "Back",
          "noSpecificAccount": "noSpecificAccount",
        },
        "deleteOnClick": null,
        "dispatch": [MockFunction],
        "id": undefined,
        "isArchived": false,
        "isRead": true,
        "mailDate": undefined,
        "mailOnClick": [Function],
        "mailSubject": "undefined ",
        "mailSummary": "Hello there",
        "message": Object {
          "account": Object {
            "number": undefined,
          },
          "getDateCreated": [MockFunction],
          "getMessageBody": [MockFunction],
          "getReference": [MockFunction],
          "getSubject": [MockFunction],
          "message": "Hello there",
          "status": "DRAFT",
        },
        "messageDetail": Object {
          "delSuccessModal": true,
          "newMessageError": true,
        },
        "messages": Object {
          "draftError": true,
        },
        "moveToInboxOnClick": null,
        "replyOnClick": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithTheme(MailItem)
          archiveOnClick={null}
          content={
            Object {
              "back": "Back",
              "noSpecificAccount": "noSpecificAccount",
            }
          }
          deleteOnClick={null}
          dispatch={[MockFunction]}
          id={undefined}
          isArchived={false}
          isRead={true}
          mailDate={undefined}
          mailOnClick={[Function]}
          mailSubject="undefined "
          mailSummary="Hello there"
          message={
            Object {
              "account": Object {
                "number": undefined,
              },
              "getDateCreated": [MockFunction],
              "getMessageBody": [MockFunction],
              "getReference": [MockFunction],
              "getSubject": [MockFunction],
              "message": "Hello there",
              "status": "DRAFT",
            }
          }
          messageDetail={
            Object {
              "delSuccessModal": true,
              "newMessageError": true,
            }
          }
          messages={
            Object {
              "draftError": true,
            }
          }
          moveToInboxOnClick={null}
          replyOnClick={null}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "archiveOnClick": null,
          "content": Object {
            "back": "Back",
            "noSpecificAccount": "noSpecificAccount",
          },
          "deleteOnClick": null,
          "dispatch": [MockFunction],
          "id": undefined,
          "isArchived": false,
          "isRead": true,
          "mailDate": undefined,
          "mailOnClick": [Function],
          "mailSubject": "undefined ",
          "mailSummary": "Hello there",
          "message": Object {
            "account": Object {
              "number": undefined,
            },
            "getDateCreated": [MockFunction],
            "getMessageBody": [MockFunction],
            "getReference": [MockFunction],
            "getSubject": [MockFunction],
            "message": "Hello there",
            "status": "DRAFT",
          },
          "messageDetail": Object {
            "delSuccessModal": true,
            "newMessageError": true,
          },
          "messages": Object {
            "draftError": true,
          },
          "moveToInboxOnClick": null,
          "replyOnClick": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`SecureMessageSummary snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SecureMessageSummary
    content={
      Object {
        "back": "Back",
        "noSpecificAccount": "noSpecificAccount",
      }
    }
    dispatch={[MockFunction]}
    message={
      Object {
        "account": Object {
          "number": "32236",
        },
        "getDateCreated": [MockFunction],
        "getMessageBody": [MockFunction],
        "getReference": [MockFunction],
        "getSubject": [MockFunction],
        "message": "Hello there",
        "status": "NEW",
      }
    }
    messageDetail={
      Object {
        "delSuccessModal": true,
        "newMessageError": true,
      }
    }
    messages={
      Object {
        "draftError": true,
      }
    }
    viewMessageFlag={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithTheme(MailItem)
        archiveOnClick={null}
        content={
          Object {
            "back": "Back",
            "noSpecificAccount": "noSpecificAccount",
          }
        }
        deleteOnClick={null}
        dispatch={[MockFunction]}
        id={undefined}
        isArchived={false}
        isRead={false}
        mailDate={undefined}
        mailOnClick={[Function]}
        mailSubject="undefined "
        mailSummary="Hello there"
        message={
          Object {
            "account": Object {
              "number": "32236",
            },
            "getDateCreated": [MockFunction],
            "getMessageBody": [MockFunction],
            "getReference": [MockFunction],
            "getSubject": [MockFunction],
            "message": "Hello there",
            "status": "NEW",
          }
        }
        messageDetail={
          Object {
            "delSuccessModal": true,
            "newMessageError": true,
          }
        }
        messages={
          Object {
            "draftError": true,
          }
        }
        moveToInboxOnClick={null}
        replyOnClick={null}
        viewMessageFlag={true}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "archiveOnClick": null,
        "content": Object {
          "back": "Back",
          "noSpecificAccount": "noSpecificAccount",
        },
        "deleteOnClick": null,
        "dispatch": [MockFunction],
        "id": undefined,
        "isArchived": false,
        "isRead": false,
        "mailDate": undefined,
        "mailOnClick": [Function],
        "mailSubject": "undefined ",
        "mailSummary": "Hello there",
        "message": Object {
          "account": Object {
            "number": "32236",
          },
          "getDateCreated": [MockFunction],
          "getMessageBody": [MockFunction],
          "getReference": [MockFunction],
          "getSubject": [MockFunction],
          "message": "Hello there",
          "status": "NEW",
        },
        "messageDetail": Object {
          "delSuccessModal": true,
          "newMessageError": true,
        },
        "messages": Object {
          "draftError": true,
        },
        "moveToInboxOnClick": null,
        "replyOnClick": null,
        "viewMessageFlag": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithTheme(MailItem)
          archiveOnClick={null}
          content={
            Object {
              "back": "Back",
              "noSpecificAccount": "noSpecificAccount",
            }
          }
          deleteOnClick={null}
          dispatch={[MockFunction]}
          id={undefined}
          isArchived={false}
          isRead={false}
          mailDate={undefined}
          mailOnClick={[Function]}
          mailSubject="undefined "
          mailSummary="Hello there"
          message={
            Object {
              "account": Object {
                "number": "32236",
              },
              "getDateCreated": [MockFunction],
              "getMessageBody": [MockFunction],
              "getReference": [MockFunction],
              "getSubject": [MockFunction],
              "message": "Hello there",
              "status": "NEW",
            }
          }
          messageDetail={
            Object {
              "delSuccessModal": true,
              "newMessageError": true,
            }
          }
          messages={
            Object {
              "draftError": true,
            }
          }
          moveToInboxOnClick={null}
          replyOnClick={null}
          viewMessageFlag={true}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "archiveOnClick": null,
          "content": Object {
            "back": "Back",
            "noSpecificAccount": "noSpecificAccount",
          },
          "deleteOnClick": null,
          "dispatch": [MockFunction],
          "id": undefined,
          "isArchived": false,
          "isRead": false,
          "mailDate": undefined,
          "mailOnClick": [Function],
          "mailSubject": "undefined ",
          "mailSummary": "Hello there",
          "message": Object {
            "account": Object {
              "number": "32236",
            },
            "getDateCreated": [MockFunction],
            "getMessageBody": [MockFunction],
            "getReference": [MockFunction],
            "getSubject": [MockFunction],
            "message": "Hello there",
            "status": "NEW",
          },
          "messageDetail": Object {
            "delSuccessModal": true,
            "newMessageError": true,
          },
          "messages": Object {
            "draftError": true,
          },
          "moveToInboxOnClick": null,
          "replyOnClick": null,
          "viewMessageFlag": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
